include/aws/common/byte_buf.h

```diff
@@ -464,146 +470,28 @@ int aws_byte_cursor_compare_lookup(
+
+AWS_COMMON_API struct aws_byte_cursor aws_byte_cursor_from_buf(const struct aws_byte_buf *const buf);
+
+AWS_COMMON_API struct aws_byte_cursor aws_byte_cursor_from_c_str(const char *c_str);
+
+AWS_COMMON_API struct aws_byte_cursor aws_byte_cursor_from_array(const void *const bytes, const size_t len);
+
 /**
  * If index >= bound, bound > (SIZE_MAX / 2), or index > (SIZE_MAX / 2), returns
  * 0. Otherwise, returns UINTPTR_MAX.  This function is designed to return the correct
  * value even under CPU speculation conditions, and is intended to be used for
  * SPECTRE mitigation purposes.
  */
-AWS_STATIC_IMPL size_t aws_nospec_mask(size_t index, size_t bound) {
-    /*
-     * SPECTRE mitigation - we compute a mask that will be zero if len < 0
-     * or len >= buf->len, and all-ones otherwise, and AND it into the index.
-     * It is critical that we avoid any branches in this logic.
-     */
-
-    /*
-     * Hide the index value from the optimizer. This helps ensure that all this
-     * logic doesn't get eliminated.
-     */
-#if defined(__GNUC__) || defined(__clang__)
-    __asm__ __volatile__("" : "+r"(index));
-#endif
-#if defined(_MSVC_LANG)
-    /*
-     * MSVC doesn't have a good way for us to blind the optimizer, and doesn't
-     * even have inline asm on x64. Some experimentation indicates that this
-     * hack seems to confuse it sufficiently for our needs.
-     */
-    *((volatile uint8_t *)&index) += 0;
-#endif
-
-    /*
-     * If len > (SIZE_MAX / 2), then we can end up with len - buf->len being
-     * positive simply because the sign bit got inverted away. So we also check
-     * that the sign bit isn't set from the start.
-     *
-     * We also check that bound <= (SIZE_MAX / 2) to catch cases where the
-     * buffer is _already_ out of bounds.
-     */
-    size_t negative_mask = index | bound;
-    size_t toobig_mask = bound - index - (uintptr_t)1;
-    size_t combined_mask = negative_mask | toobig_mask;
-
-    /*
-     * combined_mask needs to have its sign bit OFF for us to be in range.
-     * We'd like to expand this to a mask we can AND into our index, so flip
-     * that bit (and everything else), shift it over so it's the only bit in the
-     * ones position, and multiply across the entire register.
-     *
-     * First, extract the (inverse) top bit and move it to the lowest bit.
-     * Because there's no standard SIZE_BIT in C99, we'll divide by a mask with
-     * just the top bit set instead.
-     */
-
-    combined_mask = (~combined_mask) / (SIZE_MAX - (SIZE_MAX >> 1));
-
-    /*
-     * Now multiply it to replicate it across all bits.
-     *
-     * Note that GCC is smart enough to optimize the divide-and-multiply into
-     * an arithmetic right shift operation on x86.
-     */
-    combined_mask = combined_mask * UINTPTR_MAX;
-
-    return combined_mask;
-}
-#ifdef CBMC
-#    pragma CPROVER check pop
-#endif
+AWS_COMMON_API size_t aws_nospec_mask(size_t index, size_t bound);
```

Comment:
Do we actually need to export this? Pretty sure we just use this internally.

Reply:
Move it to a private header, or leave it in the public header but without the ```AWS_COMMON_API```?

Reply:
As per offline discussion, move to private header.